Automattic/jetpack
# Comments: Moderation Emails Overrides

Related:

https://github.com/Automattic/wp-calypso/issues/19110
https://github.com/Automattic/wp-calypso/issues/17949

Depends on:

https://github.com/Automattic/wp-calypso/pull/19635

#### Changes proposed in this Pull Request:

* Overrides the pluggable functions `wp_notify_postauthor` and `wp_notify_moderator` to point the user to WordPress.com
* The code has been taken verbatim from `pluggable.php` on Core `v4.8.3` and only the moderation links have been modified.
* Should only work on active and connected instances of Jetpack.
* Should only work when all the users to be notified are connected to WordPress.com

#### Testing instructions:

Enable `edit_links_calypso_redirect` option by adding `true` to [this line](https://github.com/Automattic/jetpack/pull/8144/files#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R641):

```js
if ( JetpackOptions::get_option( 'edit_links_calypso_redirect', true ) && ! is_admin() ) {...}
```

To test moderation

* Set _Comment must be manually approved_ on  _Settings_ > _Discussion_
* Verify that the admin user is connected to WordPress.com.
* Comment on any post
* The moderation email notifications links should point to WordPress.com and should include the correct site slug and action.

To test notifications

* Disable Set _Comment must be manually approved_ on  _Settings_ > _Discussion_
* Verify that the author of the post to comment is connected to WordPress.com.
* Comment on a post.
* The email notifications links should point to WordPress.com and should include the correct site slug and action.
